File: apps/cargo-tracker/app/cargotracker/handling/domain/model/repositories/HandlingActivityRepository.scala
package cargotracker.handling.domain.model.repositories

import cargotracker.handling.domain.model.aggregates.HandlingActivity

trait HandlingActivityRepository:
  def save(activity: HandlingActivity): Unit
  def findByBookingId(bookingId: String): Seq[HandlingActivity]
  def findAll(): Seq[HandlingActivity]

File: apps/cargo-tracker/src/contexts/handling/domain/repository/handling-activity-repository.ts
import type { HandlingActivity } from '../model/handling-activity.js';

/** 荷役作業リポジトリ（出力ポート） */
export interface HandlingActivityRepository {
  save(activity: HandlingActivity): Promise<HandlingActivity>;
  findByBookingId(bookingId: string): Promise<HandlingActivity[]>;
}
